horace_core/sqw/page_operations/PageOp_split_sqw.m

```diff
@@ -0,0 +1,214 @@
+                this_img.e(npix_idx(1):npix_idx(2))    = ...
+                    this_img.e(npix_idx(1):npix_idx(2)) + e_ar(:);
+                this_img.npix(npix_idx(1):npix_idx(2))    = ...
+                    this_img.npix(npix_idx(1):npix_idx(2)) + obj_npix(:);
+                % assign modified data back operation holder
+                obj.out_img{splitobj_num} = this_img;
+                obj.out_pix{splitobj_num}  = ...
+                    obj.out_pix{splitobj_num}.set_raw_data(obj_pix);
+            end
+        end
+
+        function obj = common_page_op(obj)
+            % Method contains split_sqw-specific code which runs for any
+            % page operation.
+            %
+            % Input:
+            % obj   -- pageOp object, containing pixel_data page
+            %          to split into sub-pages.
+            %
+
+            % number of objects which contain pixels
+            n_contr = numel(obj.n_obj_contrib_to_page_);
+            for i=1:n_contr
+                splitobj_num = obj.n_obj_contrib_to_page_(i);
+                % reevaluate pix ranges for every object contributing to
+                % page
+                pix_obj_data = obj.out_pix{splitobj_num}.data;
+                obj.obj_pix_ranges{splitobj_num} = ...
+                    PixelData.pix_minmax_ranges(pix_obj_data, ...
+                    obj.obj_pix_ranges{splitobj_num});
+                % store split result into places, specific for each split object
+                obj.out_pix{splitobj_num} = ...
+                    obj.out_pix{splitobj_num}.store_page_data( ...
+                    pix_obj_data,obj.write_handles{splitobj_num});
+            end
+        end
+        function [out_obj,obj] = finish_op(obj,varargin)
+            % Finalize page operations. Specific to split_sqw as the result
+            % here is very different from the majority of PageOp
+            %
+            % Input:
+            % obj     -- instance of the page operations
+            %
+            % Returns:
+            % out_obj -- array of output objects
+            % obj     -- nullified PageOp object.
+
+            obj.out_img       = obj.out_img(obj.run_contributes_);
+            obj.out_pix       = obj.out_pix(obj.run_contributes_);
+            obj.out_sqw       = obj.out_sqw(obj.run_contributes_);
+            obj.obj_pix_ranges= obj.obj_pix_ranges(obj.run_contributes_);
+            % this will delete unused handles (no contribution) and
+            % correspondent files, if some were initialized before.
+            obj.write_handles = obj.write_handles(obj.run_contributes_);
+            % Allocate resulting sqw array
+            n_obj = numel(obj.out_img);
+            out_obj = repmat(sqw,[1,n_obj]);
+            for i=1:n_obj
+                split_img =  obj.out_img{i};
+                split_pix =  obj.out_pix{i};
+                % Set single split object as the result of page operation
```

Comment:
You aren't setting a "single split object" are you? Isn't this an array of objects.

Reply:
This is single object and I am finalizing every single object within the loop.
Producing array of objects.